R/locations_and_mascots.R

```diff
@@ -0,0 +1,152 @@
+## LOCATIONS AND MASCOTS (by: pep)
+## FOR: MLB, NBA, NCAA, NFL, and NHL (NOT: EPL or MLS)
+
+## please run the ncaa_color_update script
+## please run the logos.R script
+## make sure to load rvest after tidyverse (purr)
+
+library(tidyverse)
+library(rvest) 
+
+
+## Removing epl and mls leagues
+
+teamcolors_without_epl_mls <- teamcolors %>%
+  filter(league %in% c("mlb","nba","nfl","nhl","ncaa"))
+
+## this is a function that is called on in the name_location_mascot_function below
+## the only thing that this function does is replaces all of the specificed strings to include an underscore
+## this allows the main function to seperate the name of the team into location and mascot, using pluck 
+
+nasty_gsub_function <- function(df){
+  
+  df <- mutate(df,
+               
+               ## Taking care of the spaces after the first word of a name that has at least two words
+               ## Don't use any state news, or it will replace underscores for the spaces of all colleges in that state
+               
+               name = gsub("^East ", "East_", name),
+               name = gsub("^Eastern ", "Eastern_", name),
+               name = gsub("^George ", "George_", name),
+               name = gsub("^Long ", "Long_", name),
+               name = gsub("^Los ", "Los_", name),
+               name = gsub("^New ", "New_", name),
+               name = gsub("^North ", "North_", name),
+               name = gsub("^Northern ", "Northern_", name),
+               name = gsub("^San ", "San_", name),
+               name = gsub("^St. ", "St._", name),
+               name = gsub("^Saint ", "Saint_", name),
+               name = gsub("^South ", "South_", name),
+               name = gsub("^Southern ", "Southern_", name),
```

Comment:
If you want to get really crazy, you could even try something like this `gsub("^(East|North|South|West){ern}? ", "\1_", name)`. I'm not sure I have that exactly right, but I idea would be to capture a bunch of use cases together. 

Feel free to ignore this.